pkg/metricprovider/custom_metric_provider.go

```diff
@@ -229,7 +235,9 @@ func GetPrediction(ctx context.Context, kubeclient client.Client, namespace stri
 	matchingLabels := client.MatchingLabels(map[string]string{"app.kubernetes.io/managed-by": known.EffectiveHorizontalPodAutoscalerManagedBy})
 	// merge metric selectors
 	for key, value := range labelSelector {
-		matchingLabels[key] = value
+		if key == "targetName" {
```

Comment:
what's this works for?

Reply:
because labelSelector set from autoscaling.crane.io/effective-hpa-uid to 
					Selector: &metav1.LabelSelector{
						MatchLabels: map[string]string{
							"targetKind":         ehpa.Spec.ScaleTargetRef.Kind,
							"targetName":         ehpa.Spec.ScaleTargetRef.Name,
							"targetNamespace":    ehpa.Namespace,
							"resourceIdentifier": metricIdentifier,
						},
					},

Reply:
from parameter we get metricSelector, which need to matched prediction labels,
			Labels: map[string]string{
				"app.kubernetes.io/name":                       name,
				"app.kubernetes.io/part-of":                    ehpa.Name,
				"app.kubernetes.io/managed-by":                 known.EffectiveHorizontalPodAutoscalerManagedBy,
				known.EffectiveHorizontalPodAutoscalerUidLabel: string(ehpa.UID),
			},
so we can changed effective-hpa-uid to "app.kubernetes.io/part-of":                    ehpa.Name
we can also add a label like ["app.kubernetes.io/target-name":                ehpa.OwnerReferences[0].Name]
This will be more accurate

Reply:
https://github.com/gocrane/crane/pull/520/commits/01a528d43c07389319b344ce4d30660238690cb8